internal/controllers/topology/cluster/patches/variables/variables_test.go

```diff
@@ -34,13 +34,16 @@ import (
 
 func TestGlobal(t *testing.T) {
 	tests := []struct {
-		name            string
-		clusterTopology *clusterv1.Topology
-		cluster         *clusterv1.Cluster
-		want            []runtimehooksv1.Variable
+		name             string
+		clusterTopology  *clusterv1.Topology
+		cluster          *clusterv1.Cluster
+		forPatch         string
+		validDefinitions map[string]bool
+		want             []runtimehooksv1.Variable
 	}{
 		{
-			name: "Should calculate global variables",
+			name:             "Should calculate global variables",
```

Comment:
Not sure if it makes sense to have a test case where forPatch is empty as we never pass in an empty string.

But looks like the behavior of the func is that if `forPatch` is empty we only get the "global" variables.

Hm, I think I would prefer just returning an error if definitionFrom is empty vs making this an supported case by the func that we just never use

(similar for the MD func / test)

Reply:
Fair enough - I don't necessarily want to add this error case for each function, but I have added it at the higher level function.

I've added in a patchName for these, but there shouldn't be any behaviour change.

Reply:
Sounds good